kinesis-produce-consume/src/main/java/demo/stream/OrderStreamConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package demo.stream;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+
+import demo.repository.OrderRepository;
+
+/**
+ * 
+ * @author Peter Oates
+ *
+ */
+@EnableBinding(OrderProcessor.class)
+public class OrderStreamConfiguration {
+
+	private final Log logger = LogFactory.getLog(getClass());
+	
+    @Autowired 
+    private OrderRepository orders;
+	
+	  @StreamListener(OrderProcessor.INPUT)
+	  public void processOrder(Event event) {
+
+		  //log the order received
+		  if (!event.getOriginator().equals("KinesisProducer")) { 
+			  logger.info("An order has been received " + event.toString());
+			  
+			  orders.save(event.getSubject());
```

Comment:
Hm. OK. I may live with the JPA in this sample, but let's reconsider the same `orders.save()` in the `OrderController`!
I mean remove it from the altogether, since it really makes more sense exactly here after consuming from the Kinesis.

Reply:
I did consider removing JPA to simplify the sample. The way I've been using it in testing though is to run 2 versions, one to produce messages and one to receive and save them. Then I compare the in memory dbs to confirm both contain the same Orders. 

So I could clarify this and make this a more complex sample or I could simplify and remove JPA from this sample. What do you think?

Reply:
+1 for just an additional explanation in the README.
Let's live the JPA part in this example! Not too much code anyway! 😄 

Sorry for noise BTW